pkg/kube/qdr/tls_handler.go

```diff
@@ -0,0 +1,206 @@
+package qdr
+
+import (
+	"fmt"
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/kube"
+	"github.com/skupperproject/skupper/pkg/qdr"
+	"github.com/skupperproject/skupper/pkg/service"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"strings"
+)
+
+type TlsManagerInterface interface {
+	GetSecret(string) (*corev1.Secret, error)
+	GetConfigMap() (*corev1.ConfigMap, error)
+	NewSecret(types.Credential, *metav1.OwnerReference) (*corev1.Secret, error)
+	DeleteSecret(string) error
+	AddSslProfile(sslProfile qdr.SslProfile) error
+	ExistsSslProfile(string) (bool, error)
+	RemoveSslProfile(string) error
+}
+
+type TlsServiceSupport struct {
+	Address       string
+	Credentials   string
+	CertAuthority string
+}
+
+func EnableTlsSupport(support TlsServiceSupport, tlsManager TlsManagerInterface) error {
+
+	if support.Credentials != "" {
+		if isGeneratedBySkupper(support.Credentials) {
+			// If the requested certificate is one generated by skupper it can be generated in other sites as well
+			_, err := tlsManager.GetSecret(support.Credentials)
+			if err != nil {
+				serviceSecret, err := generateNewSecret(support, tlsManager)
+				if err != nil {
+					return err
+				}
+
+				return includeSslProfile(qdr.SslProfile{Name: serviceSecret.Name}, tlsManager)
+			}
+		} else {
+			_, err := tlsManager.GetSecret(support.Credentials)
+			if err != nil {
+				return fmt.Errorf("The secret %s for address %s is missing", support.Credentials, support.Address)
+			}
+
+			err = checkAndIncludeSslProfile(qdr.SslProfile{Name: support.Credentials}, tlsManager)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if support.CertAuthority != "" && support.CertAuthority != types.ServiceClientSecret {
+		_, err := tlsManager.GetSecret(support.CertAuthority)
+		if err != nil {
+			return fmt.Errorf("The secret %s for address %s is missing", support.CertAuthority, support.Address)
+		}
+
+		sslProfile := qdr.SslProfile{
+			Name:       support.CertAuthority,
+			CaCertFile: fmt.Sprintf("/etc/skupper-router-certs/%s/ca.crt", support.CertAuthority),
+		}
+
+		err = checkAndIncludeSslProfile(sslProfile, tlsManager)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func DisableTlsSupport(credentials string, tlsManager TlsManagerInterface) error {
```

Comment:
Also suggest adding to TlsManagerInterface

Reply:
Done!